actors/builtin/market/market_actor.go

```diff
@@ -320,24 +286,24 @@ func (a Actor) ActivateDeals(rt Runtime, params *ActivateDealsParams) *adt.Empty
 	var st State
 	store := adt.AsStore(rt)
 
-	// Update deal states.
+	// Update deal dealStates.
 	rt.State().Transaction(&st, func() interface{} {
 		_, _, err := ValidateDealsForActivation(&st, store, params.DealIDs, minerAddr, params.SectorExpiry, currEpoch)
-		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to validate proposals for activation")
-
-		states, err := AsDealStateArray(store, st.States)
-		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load deal states")
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to validate dealProposals for activation")
 
 		pending, err := adt.AsMap(adt.AsStore(rt), st.PendingProposals)
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "load pending %v", err)
 
 		proposals, err := AsDealProposalArray(adt.AsStore(rt), st.Proposals)
```

Comment:
Again if the flush doesn't incur any writes I get a sense that it is cleaner and safer to put those objects on the msm too.

Reply:
And if the underlying flushes do happen to do unnecessary flushes a simple extension I would like would be to have some kind of readonly option you can set during build options.

i.e have some enum type `MarketStateMutationPermission` with `ReadOnly` and `Write` values and then: `newMarketStateMutation(&st, store).withDealStates(Write).withPending(ReadOnly).withProposals(ReadOnly)`

Reply:
Even if underlying flushes don't do I/O we could do the above to avoid wasted operations, but the time savings are likely very low.  It could be nice from a usability standpoint though, looking at the above build I know all the adts I'm reading and which ones get flushed at a glance.

Reply:
@ZenGround0 Great Idea. Making this change.